librz/core/cmd_type.c

```diff
@@ -1735,3 +1822,162 @@ RZ_IPI int rz_cmd_type(void *data, const char *input) {
 	}
 	return true;
 }
+
+RZ_IPI RzCmdStatus rz_type_cc_list_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	const char *cc = argc > 1 ? argv[1] : NULL;
+	if (cc) {
+		types_cc_print(core, cc);
+	} else {
+		types_cc_print_all(core, mode);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_type_cc_del_handler(RzCore *core, int argc, const char **argv) {
+	const char *cc = argc > 1 ? argv[1] : NULL;
+	if (!cc) {
+		eprintf("Specify the calling convention");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_analysis_cc_del(core->analysis, cc);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_type_cc_del_all_handler(RzCore *core, int argc, const char **argv) {
+	sdb_reset(core->analysis->sdb_cc);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_type_list_c_handler(RzCore *core, int argc, const char **argv) {
+	const char *ctype = argc > 1 ? argv[1] : NULL;
+	if (!ctype) {
+		type_list_c_all(core);
+		return RZ_CMD_STATUS_OK;
+	}
+	if (!print_type_c(core, ctype)) {
+		eprintf("Wrong type syntax");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_type_list_c_nl_handler(RzCore *core, int argc, const char **argv) {
+	type_list_c_all_nl(core);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_type_define_handler(RzCore *core, int argc, const char **argv) {
+	const char *type = argc > 1 ? argv[1] : NULL;
+	if (!type) {
+		eprintf("Missing type definition");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	type_define(core, type);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_type_kuery_handler(RzCore *core, int argc, const char **argv) {
+	const char *query = argc > 1 ? argv[1] : NULL;
+	Sdb *TDB = core->analysis->sdb_types;
+	char *output = NULL;
+	if (query) {
+		output = sdb_querys(TDB, NULL, -1, query);
+	} else {
+		output = sdb_querys(TDB, NULL, -1, "*");
+	}
+	if (output) {
+		rz_cons_print(output);
+		free(output);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_type_list_noreturn_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	const char *name = argc > 1 ? argv[1] : NULL;
+	if (name) {
+		ut64 n = rz_num_math(core->num, name);
+		if (n) {
+			rz_analysis_noreturn_add(core->analysis, name, n);
+		} else {
+			rz_analysis_noreturn_add(core->analysis, name, UT64_MAX);
+		}
+	} else {
+		rz_core_analysis_noreturn_print(core, mode);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_type_noreturn_del_handler(RzCore *core, int argc, const char **argv) {
+	if (argc == 1) {
+		eprintf("Specify the name or the address");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	for (int i = 1; i < argc; i++) {
+		noreturn_del(core, argv[i]);
```

Comment:
calling `noreturn_del` as is is wrong, because that function splits the input by spaces... We don't need those kind of things in newshell, also considering that an argument in newshell could even contain spaces.